docs: make |lspconfig-servers| jump to the markdown file (#1491)
neovim/nvim-lspconfig: doc/lspconfig.txt

`lspconfig` is not required to use the built-in client, it is only one front-end
interface for when a language server specific plugin is not available.

> See |lspconfig-server-configurations| for the reference to the complete list
> of language servers.

Reviewer: ```suggestion
See |lspconfig-server-configurations| for the complete list of language servers.
```

    require'lspconfig'.clangd.setup{}
<

> - create a new project, ensure it contains a root marker which matches the

Reviewer: ```suggestion
- create a new project, ensure that it contains a root marker which matches the
```

  server requirements specified in |lspconfig-server-configurations|.

- open a file within that project, such as `main.c`.

> - If you need more information about a server configuration read it's entry in

Reviewer 2: it's -> its

> - If you need more information about a server configuration read it's entry in
>   |lspconfig-server-configurations|

Reviewer: ```suggestion
- If you need more information about a server configuration, read the corresponding
  entry in |lspconfig-server-configurations|.
```

    require'lspconfig'.clangd.setup{}
<
- create a new project, ensure it contains a root marker which matches the

>   server requirements specified in |lspconfig-servers|.

Reviewer 2: lspconfig-server-configurations maybe?

  The script that automatically generates `server_configurations.md` converts
  the `package.json` referenced in a server configuration source file

>   into a list of optional settings listed in |lspconfig-server-configurations|

Reviewer: ```suggestion
  into a list of optional settings listed in |lspconfig-server-configurations|.
```

==============================================================================
SERVER CONFIGURATIONS                                 *lspconfig-configurations*

> See |lspconfig-server-configurations| by typing `K` over it for the complete

Reviewer 2: I would put this explanation in the first reference (intro) IMO

   lspconfig.foo_lsp.setup{}
<
==============================================================================

> CONFIGURATIONS                                 *lspconfig-server-configurations*

Reviewer 2: won't the tag from this conflict with the tag from the tagfile? Should we just delete this section?

contained in `:LspInfo`:

- `:LspStart <config_name>` launches the requested (configured) client, but only

>   if it successfully resolves a root directory. Note Defaults to all

Reviewer 2: I would remove note here

Reviewer 2: or add a comma/colon
